integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/apim/APIMLifecycleBaseTest.java

```diff
@@ -0,0 +1,771 @@
+
+    /**
+     * Map of the predefined application info. Key represents the application name and the value represents the
+     * Application instance.
+     */
+    private static Map<String, Application> applicationListByName = new LinkedHashMap<>();
+
+    /**
+     * Map that binds the index of the application to the application name.
+     */
+    private static Map<Integer, String> applicationListByIndex = new LinkedHashMap<>();
+
+    static {
+        // list of predefined applications.
+        addToApplicationList("SubscriptionValidationTestApp",
+                             new Application("SubscriptionValidationTestApp",
+                                             "Test Application for SubscriptionValidationTestCase",
+                                             TestConstant.APPLICATION_TIER.UNLIMITED,
+                                             ApplicationDTO.TokenTypeEnum.JWT));
+
+        // list of predefined api requests
+        String apiName = "SubscriptionValidationTestAPI";
+        String apiContext = "subscriptionValidationTestAPI";
+        String apiEndPointPostfixUrl = "/v2";
+
+        try {
+            APIRequest apiRequest = new APIRequest(apiName, apiContext,
+                                                   new URL(Utils.getDockerMockServiceURLHttp(apiEndPointPostfixUrl)));
+            String API_VERSION_1_0_0 = "1.0.0";
+            apiRequest.setVersion(API_VERSION_1_0_0);
+            apiRequest.setTiersCollection(TestConstant.API_TIER.UNLIMITED);
+            apiRequest.setTier(TestConstant.API_TIER.UNLIMITED);
+
+            APIOperationsDTO apiOperationsDTO1 = new APIOperationsDTO();
+            apiOperationsDTO1.setVerb("GET");
+            apiOperationsDTO1.setTarget("/pet/findByStatus");
+            apiOperationsDTO1.setThrottlingPolicy(TestConstant.API_TIER.UNLIMITED);
+
+            List<APIOperationsDTO> operationsDTOS = new ArrayList<>();
+            operationsDTOS.add(apiOperationsDTO1);
+            apiRequest.setOperationsDTOS(operationsDTOS);
+
+            addToAPIList(apiName, apiRequest);
+        } catch (APIManagerIntegrationTestException | MalformedURLException e) {
+            LOGGER.error("Error creating the APIRequest instance for API name: " + apiName);
+        }
+    }
+
+    /**
+     * Publish an API using REST.
+     *
+     * @param apiId                   - UUID of the API,
+     * @param publisherRestClient     - Instance of APIPublisherRestClient
+     * @param isRequireReSubscription - If publish with re-subscription required option true else false.
+     * @return HttpResponse - Response of the API Publishing activity
+     * @throws MicroGWTestException -  Exception throws by the method call of changeAPILifeCycleStatusToPublish() in
+     *                              APIPublisherRestClient.java.
+     */
+    protected HttpResponse publishAPI(String apiId, RestAPIPublisherImpl publisherRestClient,
```

Comment:
If possible, shall we move these methods to a seperate file so that these methods can be used by a different test case that does not want to extend APIMLifecycleBaseTest?

Reply:
This APIMLifecycleBaseTest class is like a utility class written to the apim related test cases. if we move these methods, we need to move 95% of the code to another class. And these methods are tightly coupled to APIM scenarios. So, if someone needs to use the method, I don't see any issue extending this class (even with a class name like this). 
So, WDYT?

Reply:
The reason to make the suggestion was because they are utility methods rather than part of the actual best test case. Wouldn't it be difficult to identify the necessary parts of the best test case if we also include the util methods  in it?